lib/controller.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Controller do
+  def main(_args) do
+    run(:welcome_screen)
+  end
+
+  def run(prompt) when prompt != "Q" do
+    UserInterface.clear_screen()
+    parse_input(prompt)
+    UserInterface.line_break()
+
+    Messages.get_prompt(:menu)
+    |> UserInterface.get_input()
+    |> run()
+  end
+
+  def run("Q") do
+    execute_command("Q")
+  end
+
+  def parse_input(input) do
+    if Enum.member?(["G", "V", "I", :welcome_screen], input) do
+      Messages.get_prompt(input)
+      |> UserInterface.display()
+
+      execute_command(input)
+    else
+      Messages.get_prompt(:unknown)
+      |> UserInterface.display()
+
+      UserInterface.line_break()
+    end
+  end
+
+  def execute_command(input) do
```

Comment:
Let's think a little bit about the SOLID here. How can we make this more open to extension, but closed to modification. HINT: we're working in a functional paradigm. Don't have to fix it now, but think about it for the future.

Reply:
The other question to pose is, why am I flagging this now? Is there a code smell that's starting to stink a bit.

Reply:
Got it. Two code smells I'm seeing here: 1) a bunch of functions that take the same argument. 2) This function should be decoupled from input by parse_input/1, but is in fact still interacting with input pretty directly, and thus needs to know a lot about the expected format of the input. Just removed parse_input/1 and am working on how to make this function know less about the details of the message it expects.